### PR TITLE
238 OpenType example

### LINKED ISSUE #238: Add OpenType features example
A single example that uses as many of these as possible:

```swift
case numberSpacing(NumberSpacing)
case numberCase(NumberCase)

case superscript(Bool)
case `subscript`(Bool)
case ordinals(Bool)
case scientificInferiors(Bool)

case smallCaps(SmallCaps)

case stylisticAlternates(StylisticAlternates)
case contextualAlternates(ContextualAlternates)
```

### PR DESCRIPTION
**Pull request by ZevEisenberg**
Closes #238. Thanks to @msluetz and Sandy Machado for design feedback.

### CODE REVIEW

File: Example-iOS/DemoStrings.swift [hunk 1]
```diff
@@ -248,28 +248,85 @@ enum DemoStrings {
...
-                .numberCase(.lower))),
-        ]
+        let digits = "\n0123456789"
+
+        return NSAttributedString.composed(of: [
+            "Number Styles".styled(with: garamondStyle, .smallCaps(.fromLowercase), .color(.raizlabsRed)),
+            digits.styled(with: garamondStyle, .numberCase(.lower), .numberSpacing(.monospaced)),
+            digits.styled(with: garamondStyle, .numberCase(.upper), .numberSpacing(.monospaced)),
+            digits.styled(with: garamondStyle, .numberCase(.lower), .numberSpacing(.proportional)),
+            digits.styled(with: garamondStyle, .numberCase(.upper), .numberSpacing(.proportional))
```
Comment (Imperiopolis) [inline Example-iOS/DemoStrings.swift]: Nice to see the new API in use here :)

File: Example-iOS/DemoStrings.swift [hunk 2]
```diff
@@ -248,28 +248,85 @@ enum DemoStrings {
...
+            .font(.systemFont(ofSize: 18)),
+            .adapt(.control)
+        )
+
+        let password = "68Il14"
+        let string = "My password, <callout>\(password)</callout>, is much easier to read with stylistic alternate set six enabled: <password>\(password)</password>."
+
+        let callout = StringStyle(.color(.raizlabsRed))
+        return string.styled(with: systemFontStyle.byAdding(
+            .xmlRules([
```
Comment (Imperiopolis) [inline Example-iOS/DemoStrings.swift]: `return string.styled(with: systemFontStyle, .xmlRules(...))` (dropping the `byAdding`) might be clearer here?

File: Example-iOS/DemoStrings.swift (hunk 1 above)
Comment (ZevEisenberg) [inline Example-iOS/DemoStrings.swift]: Yeah! It seems so obvious now 😆

File: Example-iOS/DemoStrings.swift (hunk 2 above)
Comment (ZevEisenberg) [inline Example-iOS/DemoStrings.swift]: Yes, although Xcode does kind of gross things with auto-indenting.

Either A:

```swift
return string.styled(with: systemFontStyle,
                     .xmlRules([
                        .style("callout", callout),
                        .style("password", callout.byAdding(.stylisticAlternates(.six(on: true)))),
                        ])
```

Or B:

```swift
return string.styled(
    with:
    systemFontStyle,
    .xmlRules([
        .style("callout", callout),
        .style("password", callout.byAdding(.stylisticAlternates(.six(on: true)))),
        ])
)
```

Comment (ZevEisenberg) [inline Example-iOS/DemoStrings.swift]: Scratch that, the version that you literally just suggested works fine:

```swift
return string.styled(with: systemFontStyle, .xmlRules([
    .style("callout", callout),
    .style("password", callout.byAdding(.stylisticAlternates(.six(on: true)))),
    ])
)
```

**Pull request merged**